src/main/java/net/vulkanmod/render/chunk/DrawBuffers.java

```diff
@@ -200,73 +242,24 @@ private static void fakeIndirectCmd(VkCommandBuffer commandBuffer, IndirectBuffe
         }
     }
 
-    public void buildDrawBatchesDirect(StaticQueue<RenderSection> queue, Pipeline pipeline, RenderType renderType, double camX, double camY, double camZ) {
-        TerrainRenderType terrainRenderType = TerrainRenderType.get(renderType);
-        terrainRenderType.setCutoutUniform();
+    public void buildDrawBatchesDirect(StaticQueue<DrawParameters> queue, TerrainRenderType terrainRenderType, double camX, double camY, double camZ, long layout) {
+        if(queue==null || queue.size() == 0) return;
```

Comment:
You *could* use an [Optional](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) here .

Reply:
Thanks for the suggestion: Think i'l just do a minor draw code cleanup commit instead though if that's OK


(There's some duplicate code which can be extracted as a common part: this will use an AND condition which won't work as an Optional afaik)

```c++
if(typedSectionQueue!=null && typedSectionQueue.size() != 0) {
    chunkArea.drawBuffers().bindBuffers(terrainRenderType, commandBuffer, camX, camY, camZ, layout);
    if (indirectDraw) chunkArea.drawBuffers().buildDrawBatchesIndirect(indirectBuffers[currentFrame], typedSectionQueue, terrainRenderType);
    else chunkArea.drawBuffers().buildDrawBatchesDirect(typedSectionQueue, terrainRenderType);
}
```